lib/Frontend/ParseableInterfaceModuleLoader.cpp

```diff
@@ -356,48 +356,36 @@ class swift::ParseableInterfaceBuilder {
     return false;
   }
 
-  /// Populate the provided \p Deps with \c FileDependency entries including:
-  ///
-  ///    - All the dependencies mentioned by \p SubInstance's DependencyTracker,
-  ///      that were read while compiling the module.
-  ///
-  ///    - For any file in the latter set that is itself a .swiftmodule
-  ///      living in \p cachePath, all of _its_ dependencies, copied
-  ///      out to avoid having to do recursive scanning when rechecking this
-  ///      dependency in the future.
+  /// Determines if the dependency with the provided path is a swiftmodule in
+  /// either the module cache or prebuilt module cache
+  bool isCachedModule(StringRef DepName) const {
+    if (moduleCachePath.empty() && prebuiltCachePath.empty())
+        return false;
+
+    auto Ext = llvm::sys::path::extension(DepName);
+    auto Ty = file_types::lookupTypeForExtension(Ext);
+    return Ty == file_types::TY_SwiftModuleFile &&
+        ((!moduleCachePath.empty() && DepName.startswith(moduleCachePath)) ||
+         (!prebuiltCachePath.empty() && DepName.startswith(prebuiltCachePath)));
```

Comment:
Nitpick: I think this might be clearer as multiple `if` statements, rather than trying to make one big expression.